pykokkos/lib/ufuncs.py

```diff
@@ -1,33 +1,28 @@
+from functools import singledispatch
+
 import pykokkos as pk
 
 # NOTE: it might be slick if we could encapsulate
-# the workload code inside of the ufunc definitions,
+# the workunit code inside of the ufunc definitions,
 # but we currently require global scoping
 
 
-@pk.workload
-class ReciprocalView1D:
-    def __init__(self, view: pk.View1D[pk.double]):
-        # TODO: why doesn't view have a "size"
-        # attribute? it is basically the shape product
-        # in higher dims
-        self.threads: int = view.shape[0] # type: ignore
-        self.view: pk.View1D[pk.double] = view
+@pk.workunit
+def reciprocal_impl_1d_double(view: pk.View1D[pk.double], tid: int):
+    view[tid] = 1 / view[tid] # type: ignore
 
-    @pk.main
-    def run(self) -> None:
-        pk.parallel_for(self.threads, self.pfor)
+@pk.workunit
+def reciprocal_impl_1d_single(view: pk.View1D[pk.single], tid: int):
```

Comment:
The type you're looking for is probably `pk.float`, `single` refers to a function